apps/site/lib/site_web/controllers/schedule/line/helpers.ex

```diff
@@ -71,23 +71,79 @@ defmodule SiteWeb.ScheduleController.Line.Helpers do
   def get_branch_route_stops(route, direction_id, route_pattern_id) do
     route
     |> do_get_branch_route_stops(direction_id, route_pattern_id)
-    |> RouteStop.list_from_route_patterns(route, direction_id)
-    |> Enum.chunk_by(& &1.branch)
+    |> Enum.map(&RouteStop.list_from_route_pattern(&1, route, direction_id))
+    |> make_trunks_consistent(route)
     |> RouteStops.from_route_stop_groups()
   end
 
   @spec do_get_branch_route_stops(Route.t(), direction_id(), RoutePattern.id_t() | nil) :: [
           {RoutePattern.t(), [Stop.t()]}
         ]
   defp do_get_branch_route_stops(route, direction_id, route_pattern_id) do
-    route_patterns = get_line_route_patterns(route.id, direction_id, route_pattern_id)
-
-    route_patterns
+    route.id
+    |> get_line_route_patterns(direction_id, route_pattern_id)
     |> filtered_by_typicality(route, route_pattern_id)
     |> Enum.map(&stops_for_route_pattern/1)
     |> maybe_use_overarching_branch()
   end
 
+  @routes_with_trunk_discrepancies ~w(CR-Franklin CR-Providence)
+  @spec make_trunks_consistent([[RouteStop.t()]], Route.t()) :: [[RouteStop.t()]]
+  defp make_trunks_consistent(route_stop_lists, %Route{id: route_id})
+       when route_id in @routes_with_trunk_discrepancies do
+    shared_ids = shared_ids(route_stop_lists)
+
+    route_stop_lists_with_trunk_ranges =
+      Enum.map(route_stop_lists, fn route_stop_list ->
+        trunk_range =
+          range(route_stop_list, fn route_stop -> MapSet.member?(shared_ids, route_stop.id) end)
+
+        {route_stop_list, trunk_range}
+      end)
+
+    largest_trunk = largest_trunk(route_stop_lists_with_trunk_ranges)
+
+    Enum.map(route_stop_lists_with_trunk_ranges, fn {route_stops, trunk_range} ->
+      old_trunk = Enum.slice(route_stops, trunk_range)
+
+      if old_trunk == largest_trunk do
+        # No need to replace anything
+        route_stops
+      else
+        route_stops
+        |> Enum.reject(&Enum.member?(old_trunk, &1))
+        |> List.insert_at(trunk_range.first, largest_trunk)
+        |> List.flatten()
+      end
+    end)
+  end
+
+  defp make_trunks_consistent(route_stop_lists, _route), do: route_stop_lists
+
+  @spec range(Enum.t(), (Enum.element() -> as_boolean(term()))) :: Range.t()
+  defp range(items, fun) do
+    {min, max} =
+      items
+      |> Enum.with_index()
+      |> Enum.reduce([], fn {item, index}, acc -> if fun.(item), do: [index | acc], else: acc end)
+      |> Enum.min_max()
+
+    min..max
+  end
+
+  @spec largest_trunk([{[RouteStop.t()], Range.t()}]) :: [RouteStop.t()]
+  defp largest_trunk(route_stop_lists_with_trunk_ranges) do
+    largest_trunk_range =
+      route_stop_lists_with_trunk_ranges
+      |> Enum.map(&elem(&1, 1))
+      |> Enum.max()
```

Comment:
Technically this isn't so much largest trunk range (I am interpreting that to mean "longest"), so much as it is range with the highest indices. Is this the intent?

Reply:
I think that's correct. Let me phrase it another way to double-check…

Each trunk begins and ends at the same stops, but in between, the "largest" one hits the greatest number of stops.

I'm very happy if you have ideas for a better name for that.